test/hotspot/gtest/gc/g1/test_freeRegionList.cpp

```diff
@@ -44,7 +44,11 @@ TEST_OTHER_VM(G1FreeRegionList, length) {
 
   // Create a fake heap. It does not need to be valid, as the G1HeapRegion constructor
   // does not access it.
-  MemRegion heap(nullptr, num_regions_in_test * G1HeapRegion::GrainWords);
+  int val = 1;
+  HeapWord* ptr = reinterpret_cast<HeapWord*>(&val);
```

Comment:
The initial value for `val` doesn't matter, since we're using its address rather than value.
And the cast could be removed by changing the type of `val`, leading to something like this:
```
HeapWord val{};
HeapWord* ptr = align_up(&val, os::vm_page_size());
```

Reply:
I adjusted the coding .